src/query/union/simple_union.rs

```diff
@@ -0,0 +1,113 @@
+use crate::docset::{DocSet, TERMINATED};
+use crate::postings::Postings;
+use crate::DocId;
+
+/// A `SimpleUnion` is a `DocSet` that is the union of multiple `DocSet`.
+/// Unlike `BufferedUnion`, it doesn't do any horizon precomputation.
+///
+/// For that reason SimpleUnion is a good choice for queries that skip a lot.
+pub struct SimpleUnion<TDocSet> {
+    docsets: Vec<TDocSet>,
+    doc: DocId,
+}
+
+impl<TDocSet: DocSet> SimpleUnion<TDocSet> {
+    pub(crate) fn build(mut docsets: Vec<TDocSet>) -> SimpleUnion<TDocSet> {
+        docsets.retain(|docset| docset.doc() != TERMINATED);
+        let mut docset = SimpleUnion { docsets, doc: 0 };
+
+        docset.update_current();
+
+        docset
+    }
+
+    fn update_current(&mut self) {
+        let mut next_doc = TERMINATED;
+
+        for docset in &mut self.docsets {
+            next_doc = next_doc.min(docset.doc());
+        }
+        self.doc = next_doc;
+    }
```

Comment:
I don't think the name is good, and the `&mut docsets` is useful,

```suggestion
    fn initialize_first_doc_id(&mut self) -> DocId {
        let mut next_doc = TERMINATED;

        for docset in &self.docsets {
            next_doc = next_doc.min(docset.doc());
        }
        self.doc = next_doc;
    }
```